Automattic/jetpack
# Jetpack and Network: add a filter for the jetpack_disconnect capability

#### Changes proposed in this Pull Request:
 * Add a new filter for setting the `jetpack_disconnect` capability, `jetpack_disconnect_cap`.
 * Move the logic for setting the `jetpack_disconnect` capability for multisite to a new method, `Jetpack_Network::set_multisite_disconnect_cap()`, which hooks to the new filter. Also add unit tests for this method.
* Add a docblock to `Jetpack::jetpack_custom_caps()`.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This changes an existing part of Jetpack.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Confirm that you can connect and disconnect sites in a multisite installation. 
   * **NOTE**: If you’re testing using the beta plugin, you will need to manually change the `Jetpack_Network` class to recognize the dev version of Jetpack.

      In `class.jetpack-network.php`, change [line 86](https://github.com/Automattic/jetpack/blob/master/class.jetpack-network.php#L86) to this:

      `if ( is_multisite() && is_plugin_active_for_network( 'jetpack-dev/jetpack.php' ) ) {`

2. Confirm that you can connect and disconnect a single site.

#### Proposed changelog entry for your changes:
* n/a
